src/cli/cli_parser.py

```diff
@@ -0,0 +1,95 @@
+from argparse import ArgumentParser, Namespace
+
+from src.cli.types import DPRPVerPair, Settings
+from src.definitions import PackFormatsDict, DiscListEntryContents, DiscListContents
+
+from src.cli.logger import *
+from csv import reader
+
+from pathlib import Path
+
+
+def is_legacy_version(mc_ver: str) -> bool:
+    mc_ver_val: list[int] = [int(i) for i in mc_ver.split(".")]
+    return mc_ver_val <= [1, 19, 3]
+
+
+def get_dp_rp_version(mc_ver: str) -> DPRPVerPair:
+    # find the corresponding dp&rp version of the game.
+    matching_version = None
+    for version, values in PackFormatsDict.items():
+        if '-' in version:
+            start, end = version.split(' - ')
+            if start <= mc_ver <= end:
+                matching_version = version
+                break
+        elif version == mc_ver:
+            matching_version = version
+            break
+
+    if matching_version:
+        info(f"matched game version:{matching_version}")
+        dp_rp_versions = PackFormatsDict[matching_version]
+        return dp_rp_versions
+    else:
+        error(f"unknown game version{mc_ver}。")
+
+
+def absolute_path(input_path: str):
+    return str(Path(input_path).absolute())
+
+
+class CLIParser:
+    _parse_result: Namespace
+
+    # Namespace(disc_entries_file='123', minecraft_version='1.2.0', use_concurrent=False, mix_mono=False,
+    # package_name='infinite_music_discs', dest_dir='.', compress_output=False)
+
+    def __init__(self):
+        parser = ArgumentParser(prog="mc music packer")
+
+        parser.add_argument("-f", "--disc-entries-file", type=str, required=True,
+                            help="a csv file delimited by semicolon, "
+                                 "each line of which contains texture_file;track_file;title;internal_name.")
+        parser.add_argument("-g", "--minecraft-version", type=str, required=True,
+                            help="specify a game version, support 1.14 - 1.20.2",
+                            default="1.20.2")
+        parser.add_argument("-p", "--use-parallel", action='store_true',
```

Comment:
Fix: `use_parallel` implies that the user should set this flag if they want to use the parallel processing, but this flag is connected to the `skip_proc` setting which will skip the parallel processing if it's set to True. This flag should be renamed to something like `proc-serially` or `proc-sequential` so the user knows that setting it will _disable_ the parallel processing.